### DISCUSSION TITLE
How do you get the maximum electric field in a particular brain region

### DISCUSSION THREAD

**Discussion by xiberJ** [simnibs/simnibs#132] [9. Results and analysis]
Dear developers, hello

First of all, thank you very much for your software. For the simulation results, how to obtain the electric field data of a specific area, such as the maximum electric field or minimum electric field data of a small area on the surface of the brain, and the electric field data of a region inside the brain (the maximum electric field and so on)?

**Comment by oulap** [comment]
You can do a region-of-interest-based analysis using either different atlases or just using the local region around a point. Here's an example of how to span a spherical ROI around a point and calculating the average electric file magnitude within that region:

https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html

Similarly you could calculate the max on any percentile you want.

**Comment by xiberJ** [comment]
Thank you very much for your answer


If I need to take the maximum in that range,
Is ”mean_magnE = np.average(field[roi], weights=elm_vols[roi])
print('mean ', field_name, ' in M1 ROI: ', mean_magnE)“ in the code
Change to ”max_magnE = np.max(gm_surf.field[field_name][roi])
print('maximum ', field_name, ' in ', region_name, ': ', max_magnE)
“

**Comment by oulap** [comment]
Yes, taking the max should return the maximum value. You could also consider taking a percentile, see:

https://numpy.org/doc/stable/reference/generated/numpy.percentile.html

to be more robust to outlier values. For example taking a 99-%tile would ignore extreme e-field values in case there are numerical outliers.

**Comment by xiberJ** [comment]
Thank you very much for your answer

Can visualize and obtaining the inner regions of the brain of the electric field. In addition, I have seen some articles using this AAL-atlas to characterize the distribution and intensity of electric fields. How does this work?Through this study let me know a lot of interesting things, haha

**Comment by oulap** [comment]
You can extract the e-field from the mesh pretty much anywhere, you just need to define the region-of-interest. For deeper brain regions you could either look for a coordinate in MNI-space, map that to subject space, and then define a ROI around that coordinate, or you could use the structural segmentation inside the m2m_<subid>/segmentation-folder to define the ROI in subject space. I suggest you read through the analysis tutorial here:

https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html

It will describe both coordinate based ROI creation and the surface-based atlas analysis.

**Comment by xiberJ** [comment]
Thank you for your generous advice, I have a basic understanding of this code, and also obtained the size of the field value of the specified area is greater than 99 percent; I would also like to ask if the simulated electric field results can be visualized by FreeSurfer

**Comment by oulap** [comment]
Yep, when you do the simulations you can choose to map the electric field results to different spaces. In a simulation script, if you set these fields:
```
s.map_to_surf = False
s.map_to_fsavg = False
s.map_to_vol = False
s.map_to_MNI = False
```
all to True instead of False, where s is the simulation structure, the results will get mapped to the central gray matter surface of the subject (map_to_surf), to the fsaverage surface (map_to_fsavg), to the input nifti volume space (map_to_vol), and to the MNI template space (map_to_mni). You can then visualize these fields in freeview either in volume or surface space, whichever you prefer. For example if you would want to visualize the electric field on the pial surface of a subject,  you would first open the surface file inside m2m_<subid>/surfaces/<hemi>.pial.gii as a surface in freeview, then in freeview select overlay -> load generic, and then choose a file inside simnibs_simulation/subject_overlays and load that.

**Comment by xiberJ** [comment]
![1681543728697](https://user-images.githubusercontent.com/122353886/232196370-f3a8e92e-dca6-4165-b4de-b90356b756ec.png)


thanks，
How does this distribution of electric fields looking at a section of the brain work

**Comment by oulap** [comment]
Not sure where that visualization is from, but please check this tutorial:

https://simnibs.github.io/simnibs/build/html/tutorial/visualization.html

We typically use gmsh to visualize the results, if you want to look at the distribution of the e-field over a section you should look into how to do clipping in gmsh (shown in the tutorial in the link above).

**Comment by xiberJ** [comment]
Thank you very much for your patient guidance

**Reply by xiberJ** [reply]
![企业微信截图_16818176065430](https://user-images.githubusercontent.com/122353886/232764617-92e758f2-3ae9-40e0-89b9-69d06105ead9.png)
I would like to ask how this structure map of electric field distribution to the brain is realized, is this through Freesufer software

**Reply by oulap** [reply]
Those figure are most likely produced by mapping the simulation results to MNI space (on the left) and to individual space (on the right). You can do the same by setting the flags I mentioned above. This visualization doesn't look like freeview, it's probably just done in python. You can certainly visualize the e-field in a similar fashion either in freeview or fsleyes.

**Comment by Karin-Prillinger** [comment]
Hi, I tried to analyze the mean electric field in a ROI following the tutorial https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html

However, running the line below, I always receive the following error: module 'simnibs' has no attribute 'ElementTags'
gray_matter = head_mesh.crop_mesh(simnibs.ElementTags.GM). Do you maybe have any advice or ideas where I could find more information? Thanks!

**Reply by TorgeW** [reply]
Replace `simnibs.ElementTags.GM` with `2`
This is an issue with changes from the new upcoming pre-release.

**Reply by Karin-Prillinger** [reply]
Thank you!
